my-code/src/Contexts/MoviesData.tsx

```diff
@@ -0,0 +1,95 @@
+import axios from "axios";
+import { createContext, ReactNode, useContext, useEffect, useState } from "react";
+
+interface MoviesDataProviderProps {
+    children: ReactNode;
+}
+
+interface MoviesDataContextProps {
+    movies: MovieProps[];
+    getMoviesData: (search: string) => Promise<void>;
+    favouriteMovies: FavouriteMoviesProps[];
+    addToFavourites: (movieID: string) => void;
+    isFavourite: (movieID: string) => boolean;
+    isLoading: boolean;
+}
+
+interface MovieProps {
+    Actors: string;
+    Awards: string;
+    Country: string;
+    Director: string;
+    Genre: string;
+    Language: string;
+    Metascore: string;
+    Plot: string;
+    Poster: string;
+    Released: string;
+    Title: string;
+    Year: string;
+    imdbRating: string;
+    imdbID: string;
+}
+
+interface FavouriteMoviesProps {
+    imdbID: string;
+}
+
+const MoviesDataContext = createContext({} as MoviesDataContextProps)
+
+export function MoviesDataProvider({ children }: MoviesDataProviderProps) {
+
+    const [movies, setMovies] = useState<MovieProps[]>([])
+    const [favouriteMovies, setFavouriteMovies] = useState<FavouriteMoviesProps[]>(
+        JSON.parse(localStorage.getItem('favouriteMovies') || '[]')
+    )
+    const [isLoading, setIsLoading] = useState(false)
+
+    async function getMoviesData(search: string) {
+        if (search.length > 0 && search.length < 3) return;
+
+        setIsLoading(true)
+        if (search.length === 0) return (
+            setMovies([]),
+            setIsLoading(false)
+        )
+
+        const response = await axios.get(`https://www.omdbapi.com/?apikey=23fc3dfd&s=${search}`)
+        if (response.data.Response === 'False') return (
+            setMovies([]),
+            setIsLoading(false)
+        )

```

Comment:
This return is unusual. Why not do:
```suggestion
        if (response.data.Response === 'False') {
            setMovies([]),
            setIsLoading(false)
            return
        }
```
Or even: 
```suggestion
         if (response.data.Response === 'False') {
            setMovies([]),
            return setIsLoading(false)
        }
```